src/github_runner_image_builder/cli.py

```diff
@@ -94,6 +94,32 @@ def get_latest_build_id(cloud_name: str, image_name: str) -> None:
     )
 
 
+# The arguments are necessary input for click validation function.
+def _validate_juju_channel(
+    ctx: click.Context, param: click.Parameter, value: str  # pylint: disable=unused-argument
+) -> str:
+    """Validate juju channel string input.
+
+    Args:
+        ctx: Click context argument.
+        param: Click parameter argument.
+        value: The value passed into --juju option.
+
+    Raises:
+        BadParameter: If invalid juju channel was passed in.
+
+    Returns:
+        The validated Juju channel option.
+    """
+    if not value:
+        return ""
+    try:
+        [version, track] = value.strip().split("/")
```

Comment:
Why a list instead of just `version, track = ...`? 

The correct names are `<track>/<risk>`.

A channel format is a bit more complex (https://snapcraft.io/docs/channels), but I understand this is enough for this PR.

Reply:
Oh cool, thanks for the guidance - I will include this on the follow up Microk8s PR so that I can progress on the features :)